pkg/initializers/dataset/cache_initalizer.py

```diff
@@ -0,0 +1,209 @@
+        worker_mem = self.config.worker_mem or "2Gi"
+        
+        substitutions = {
+            'NAME': train_job_name,
+            'IAM_ROLE': iam_role,
+            'SIZE': cluster_size,
+            'IMAGE': cache_image,
+            'HEAD_CPU': head_cpu,
+            'HEAD_MEM': head_mem,
+            'WORKER_CPU': worker_cpu,
+            'WORKER_MEM': worker_mem,
+        }
+        
+        if self.config.metadata_loc:
+            substitutions['METADATA_LOC'] = self.config.metadata_loc
+        if self.config.table_name:
+            substitutions['TABLE_NAME'] = self.config.table_name
+        if self.config.schema_name:
+            substitutions['SCHEMA_NAME'] = self.config.schema_name
+
+        deploy_lws_with_substitution(
+            train_job_name,
+            'cache-initializer-template.yaml',
+            namespace='cache-test',
+            substitutions=substitutions,
+            timeout=600
+        )
+
+        logging.info("Cache dataset initialization completed")
+
+
+def deploy_lws_with_substitution(train_job_name, yaml_path, config_file: Optional[str] = None, namespace='default', substitutions=None,
+                                 timeout=300):
+    """
+    Deploys a parameterized LeaderWorkerSet YAML with ServiceAccount, environment substitution,
+    and waits for deployment readiness.
+
+    Args:
+        train_job_name(str): train_job_name
+        yaml_path (str): Path to YAML file containing configuration
+        namespace (str): Target Kubernetes namespace
+        substitutions (dict): Additional variables for substitution
+        timeout (int): Maximum wait time in seconds
+
+    Returns:
+        bool: True if deployment succeeded and became ready
+    """
+    with open(yaml_path, 'r') as f:
+        content = f.read()
+
+    sub_vars = {**os.environ, **(substitutions or {})}
+
+    for key, value in sub_vars.items():
+        content = content.replace(f'${key}', value)
+        content = content.replace(f'${{{key}}}', value)
+
+    resources = list(yaml.safe_load_all(content))
+
+    if config_file or not is_running_in_k8s():
+        config.load_kube_config(config_file=config_file)
+    else:
+        config.load_incluster_config()
+
+    api_client = client.ApiClient()
+    core_v1 = client.CoreV1Api(api_client)
+    custom_api = client.CustomObjectsApi(api_client)
+    training_job = custom_api.get_namespaced_custom_object(
+        group="trainer.kubeflow.org",
+        version="v1alpha1",
+        plural="trainjobs",
+        namespace=namespace,
+        name=train_job_name
+    )
+    print(f"trainJob: {training_job}")
+
+    # Create owner reference from TrainingJob
+    owner_ref = {
+        "apiVersion": training_job["apiVersion"],
+        "kind": training_job["kind"],
+        "name": training_job["metadata"]["name"],
+        "uid": training_job["metadata"]["uid"],
+        "controller": True,
+        "blockOwnerDeletion": True
+    }
```

Comment:
Can you use owner_ref = client.V1OwnerReference() object to create it, and then use V1OwnerReference.to_dict() to convert it for LWS ?

Reply:
@akshaychitneni Did you check it ?

Reply:
I found issues with converting owner_ref object to dict. For example, apiVersion gets converted to api_version